src/rebar3_hex_publish.erl

```diff
@@ -110,11 +111,18 @@ format_error({has_maintainers, AppName}) ->
 format_error({has_contributors, AppName}) ->
     Err = "~ts.app.src : deprecated field contributors found",
     io_lib:format(Err, [AppName]);
+format_error({has_unstable_deps, Deps}) ->
+    MainMsg = "The following pre-release dependencies were found : ",
+    DepList = [[Pkg] ++ " - " ++ [Ver] || {Pkg, Ver} <- Deps],
+    Msg = ["In the future packages with pre-release dependencies will be considered unstable ",
+           "and will be prevented from being published. ",
+           "We recommend you upgrade your these dependencies as soon as possible"],
+    io_lib:format("~s~n~n~s~n~n~s~n", [MainMsg, DepList, Msg]);
```

Comment:
Your DepList as a string will just look like `dep-1.0.0-rc1lib-1.1.1-rc2` and you might want to instead format it as `DepList = [io_lib:format("~s-~s ", [Pkg, Ver]) || {Pkg, Ver} <- Deps],` or something to that effect.

Reply:
I like that better, will change.